react_project/src/Chapter.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+class Chapter extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { chapnum: props.chapnum, sections: props.sections };
```

Comment:
C'est un peu bizarre d'utiliser les valeurs des props qui ne changent pas pour initialiser le state du composant qui est susceptible de changer. Surtout que vous n'utiliser ces valeurs que pour de l'affichage.

Retirer le this.state.